docs/pushpayloads.schema.json

```diff
@@ -0,0 +1,56 @@
+{
+  "title":"FxA Push Payload schema",
+  "description":"This schema defines what is acceptable to send as a payload data with the Push API from the FxA servers to a device connected to FxA",
+  "$schema":"http://json-schema.org/draft-04/schema#",
+  "type":"object",
+  "required":[
+    "version",
+    "data"
+  ],
+  "properties":{
+    "version":{
+      "type":"number",
+      "description":"The version of this push payload data instance. Bump this if you make changes to any part of this schema."
```

Comment:
Which actually means I'm starting to retread ground that we discussed way back in August 2013, in a short-lived sync-dev thread entitled "version-negotiation, how does it even work?":

  https://mail.mozilla.org/pipermail/sync-dev/2013-August/000281.html

Which is worth a read for some historical concept on the 'devices API' that we're finally getting around to building out.

From experience, this discussion has a *very* slippery slope of complexity, so I suggest we do the simplest forward-compatible thing that we possibly can.  Since integers are cheap and we're not likely to run out of them anytime soon, what about:

* Have a single integer version number for each iteration of the message schema.
* Each schema version defines the entire schema - the set of supported actions, and the payload for each action.  Add a new action?  Increment the version.  Make changes to an existing action payload?  Increment the version.
* Tell clients to ignore unknown fields in the payload.
* Then as a special exception, if you're making purely informational and optional additions to the payload of an action, and there's no way that old clients could do anything wrong, then keep the same version number.  Adding a deviceID may well fall into this category.

This would allow us to start using the version number to make runtime decisions in the future, via a scheme like:

* Have clients register their maximum supported schema version as part of their device metadata.  Clients that don't do this are assumed to be at version zero.
* Assume that clients can sensibly handle messages with format versions <= their registered maximum.
* Whenever we want to send something out, attempt to "cast" it to the highest-numbered format within the client's supported range.

Trying to push the version numbers down into the individual sub-schemas, seems like overkill to me at this stage.  And if we do find we need to get there one day, then we can declare that schema versions "X" and above introduce a new per-action version number.